src/app/(post-login)/submission/program/[shortName]/sample-registration/page.tsx

```diff
@@ -100,22 +100,82 @@ const Register = ({ shortName }: { shortName: string }) => {
 	const hasErrors = !!schemaOrValidationErrors.length;
 	const registrationId = get(clinicalRegistration, 'id', '') || '';
 
+	/**
+	 * new HTTP endpoints don't return the same objects as strictly typed gql endpoints
+	 * data shape needs to match so state can stay in sync on the frontend
+	 */
```

Comment:
Strictly this part, where we massage the API response to match the GQL shape, could be done on the backend.

Reply:
Seems like a lot of maintenance for the front end, especially considering we have functions on the Clinical side specifically intended to massage the response:

https://github.com/icgc-argo/argo-clinical/blob/develop/src/schemas/utils.ts#L258

https://github.com/icgc-argo/argo-clinical/blob/develop/src/schemas/clinical-mutations/uploadClinicalSubmissions.ts

So it might be worth reviewing what needs to change on the back end creating a ticket to address, at least to minimize future work

Reply:
I agree.

The new endpoints should not have changed the data shape or it should have been communicated & frontend state implications discussed. 

Endpoint responses could be rewritten. That would be ideal.
But the frontend is blocked until that's done without this.